data/data/azure/vnet/nsg.tf

```diff
@@ -8,20 +8,14 @@ resource "azurerm_subnet_network_security_group_association" "master" {
   count = var.preexisting_network ? 0 : 1
 
   subnet_id                 = azurerm_subnet.master_subnet[0].id
-  network_security_group_id = azurerm_network_security_group.master.id
-}
-
-resource "azurerm_network_security_group" "worker" {
-  name                = "${var.cluster_id}-node-nsg"
```

Comment:
i think you need to update, 
https://github.com/openshift/installer/blob/1df5c8ba6fa4d86c96564d9361a4cf3ab96857de/pkg/asset/manifests/cloudproviderconfig.go#L112